imports/plugins/core/accounts/server/no-meteor/mutations/addressBookAdd.js

```diff
@@ -90,13 +79,17 @@ export default async function addressBookAdd(context, address, accountUserId) {
   const result = await Accounts.updateOne({ userId }, accountsUpdateQuery);
 
   // If the address update was successful, then return the full updated address
-  if (result.modifiedCount === 1) {
-    // Find the account
-    const updatedAccount = await Accounts.findOne({ userId });
-
-    // Pull the updated address and return it
-    return updatedAccount.profile.addressBook.find((updatedAddress) => address._id === updatedAddress._id);
+  if (result.modifiedCount !== 1) {
+    throw new ReactionError("server-error", "Unable to add address to account");
   }
 
-  throw new ReactionError("server-error", "Unable to add address to account");
+  const updatedAccount = await Accounts.findOne({ userId });
+  await appEvents.emit("afterAccountUpdate", {
+    updatedAccount,
+    updatedBy: userIdFromContext,
+    updatedFields: ["profile.addressBook"]
+  });
+
+  // Pull the updated address and return it
+  return updatedAccount.profile.addressBook.find((updatedAddress) => address._id === updatedAddress._id);
```

Comment:
Do we need to check that `updatedAccount.profile.addressBook` exists here?

Reply:
Given that we just did `Accounts.updateOne` to add an address, it's very unlikely to not exist. I think the only way it wouldn't is if another process altered the account in the milliseconds between the `updateOne` and the `findOne`.

That said, zero chance of error is better than minuscule chance of error. 😄 This code is from before we started using `findOneAndUpdate`. I'll update it to use that, and then it will be guaranteed to exist.